tests/test_matmul.py

```diff
@@ -0,0 +1,155 @@
+import itertools
+
+import numpy as np
+import pytest
+
+from iree.compiler.dialects import arith, tensor, linalg
+from iree.compiler.dialects.arith import _is_float_type
+from iree.compiler.dialects.func import func
+from iree.compiler.extras import types as T
+from .conftest import invokable_module, mlir_type_to_np_dtype, ids
+
+
+@pytest.mark.parametrize("target_backend", ["amd-aie"])
+def test_smol_matmul(session_module, target_backend, device):
+    session, module = session_module
+
+    @func(T.tensor(32, 16, T.i8()), T.tensor(16, 32, T.i8()))
+    def matmul_i8_i32(lhs, rhs):
+        cst = arith.constant(T.i32(), 0)
+        v0 = tensor.empty([32, 32], T.i32())
+        v1 = linalg.fill(cst, outs=[v0])
+        return linalg.matmul(lhs, rhs, outs=[v1])
+
+    arg0 = np.ones((32, 16), dtype=np.int8)
+    arg1 = np.ones((16, 32), dtype=np.int8)
```

Comment:
Can we not use `ones` as input? If the data is rearranged incorrectly (for example through the DMAs), this will still lead to  correct results because all inputs are `1`.